server/users/authMiddleWare.js

```diff
@@ -0,0 +1,58 @@
+import { firebaseAdmin } from '../database/firestore';
+import { StatusCode } from '../utils/http';
+import { newError, sendError } from '../utils/error';
+import { getUserById } from './controller';
+
+const hasAuthHeader = (req) => {
+  return req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer';
+};
+
+const getAuthToken = (req) => {
+  return req.headers.authorization.split(' ')[1];
+};
+
+const getAuthUser = async (token) => {
+  try {
+    const firebaseUser = await firebaseAdmin.auth().verifyIdToken(token);
+    const firestoreUser = await getUserById(firebaseUser.uid);
+    return {
+      id: firebaseUser.uid,
+      ...firebaseUser,
+      ...firestoreUser,
+    };
+  } catch (e) {
+    throw newError(StatusCode.Forbidden, 'Forbidden');
+  }
+};
+
+export async function authMiddleWare(req, res, next) {
+  if (hasAuthHeader(req)) {
+    const token = getAuthToken(req);
+    try {
+      req.user = getAuthUser(token);
+
+      next();
+    } catch (e) {
+      sendError(res, e);
+    }
+  } else {
+    res.send({ message: 'Unauthorized' }).status(StatusCode.Unauthorized);
+    next(new Error('Unauthorized'));
+  }
+}
+
```

Comment:
What is the difference between `authMiddleWare` and `userMiddleWare`?

Reply:
Both add the `req.user` object, one just `throws` on missing user - in hindsight using a global with a param may have been better

Reply:
I'll push a followup